platforms/wechat/wrapper/systemInfo.js

```diff
@@ -27,7 +27,37 @@ Object.assign(adapter, {
         else {
           sys.platform = sys.WECHAT_GAME;
         }
+        // We need to set sys.platform first before calling adaptSysFunc.
+        adaptSysFunc.call(this, sys);
 
+        let loader = new Promise(function (resolve, reject) {
```

Comment:
Why add this function? Is it something with this issue?

Reply:
This is a way to check if webp is supported.
canvas.toDataUrl cannot be used.

Reply:
https://github.com/cocos/cocos-engine/pull/13165